src/ucp/core/ucp_am.c

```diff
@@ -62,17 +62,36 @@ void ucp_am_ep_init(ucp_ep_h ep)
 void ucp_am_ep_cleanup(ucp_ep_h ep)
 {
     ucp_ep_ext_proto_t *ep_ext = ucp_ep_ext_proto(ep);
+    ucp_recv_desc_t *first_rdesc, *tmp, *mid_rdesc;
+    ucs_queue_iter_t iter;
+    size_t UCS_V_UNUSED count;
 
-    if (ep->worker->context->config.features & UCP_FEATURE_AM) {
-        if (ucs_unlikely(!ucs_list_is_empty(&ep_ext->am.started_ams))) {
-            ucs_warn("worker %p: not all UCP active messages have been"
-                     " run to completion on ep %p", ep->worker, ep);
+    if (!(ep->worker->context->config.features & UCP_FEATURE_AM)) {
+        return;
+    }
+
+    if (ucs_unlikely(!ucs_list_is_empty(&ep_ext->am.started_ams))) {
```

Comment:
if I'm not mistaken we use likely/unlikely only at fast path

Reply:
fixed